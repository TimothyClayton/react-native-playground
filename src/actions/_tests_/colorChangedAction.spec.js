// import * as colorChangeActions from '../ColorChangedAction';

// describe('ColorChangeActions', () => {
//   it('should create an action to set the color', () => {
//     const colorName = { colorName: "BLUE" };
//     const expectedAction = {
//       type: 'COLOR_CHANGED',
//       payload: colorName
//     }
//     expect(colorChangeActions.colorChanged(colorName)).toEqual(expectedAction)
//   })
// })

import configureStore from 'redux-mock-store' //ES6 modules
// const { configureStore } = require('redux-mock-store') //CommonJS
import * as colorChangeActions from '../ColorChangedAction'

const middlewares = []
const mockStore = configureStore(middlewares)

const initialState = {}
const store = mockStore(initialState)

beforeEach(() => {
  store.clearActions();
})

afterEach(() => {
  expect(store.getActions()).toMatchSnapshot()
})

it('Dispatches colorChanged action', () => {
  const colorName = { colorName: "BLUE" };
  store.dispatch(colorChangeActions.colorChanged(colorName))
})
