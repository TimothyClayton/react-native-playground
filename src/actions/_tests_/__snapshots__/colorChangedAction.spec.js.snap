// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dispatch colorChanged action 1`] = `
Array [
  Object {
    "payload": Object {
      "colorName": "BLUE",
    },
    "type": "COLOR_CHANGED",
  },
]
`;

exports[`Dispatch colorChanged action 2`] = `
Array [
  Object {
    "payload": Object {
      "colorName": "BLUE",
    },
    "type": "COLOR_CHANGED",
  },
]
`;

exports[`Dispatches colorChanged action 1`] = `
Array [
  Object {
    "payload": Object {
      "colorName": "BLUE",
    },
    "type": "COLOR_CHANGED",
  },
]
`;

exports[`Dispatches colorChanged action 2`] = `
Array [
  Object {
    "payload": Object {
      "colorName": "BLUE",
    },
    "type": "COLOR_CHANGED",
  },
]
`;
